### PR TITLE
Category List API

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Fixes https://github.com/mozilla/addons-server/issues/3676, supports https://github.com/mozilla/addons-frontend/pull/1237/

### CODE REVIEW

**Comment by diox** [comment]
It's probably worth adding caching headers using `finalize_response()` in the view ? Since categories don't really change...

File: src/olympia/constants/categories.py [hunk 1]
```diff
@@ -9,21 +9,25 @@
 
 
 class StaticCategory(object):
```
Comment (diox) [inline src/olympia/constants/categories.py]: Somewhere in that class should the reason behind immutability described.

File: src/olympia/constants/categories.py [hunk 2]
```diff
@@ -9,21 +9,25 @@
...
-        self.id = id
-        self.application = app
-        self.misc = misc
-        self.name = name
-        self.slug = slug
-        self.type = type
-        self.weight = weight
+        # Avoid triggering our own __setattr__ implementation
+        # to keep immutability intact but set initial values.
+        object.__setattr__(self, 'id', id)
```
Comment (diox) [inline src/olympia/constants/categories.py]: Well, it's ugly but it does the job...

**Comment by EnTeQuAk** [comment]
Updated docs, added caching. I'm going to merge this once travis is green, if there is anything that needs fixing / improving let's do that in a separate PR. Don't want to block @tofumatt any longer.

**Pull request merged**